.github/workflows/integrationTest.yml

```diff
@@ -0,0 +1,156 @@
+          echo "$PRIVATE_KEY" > private_key && chmod 600 private_key
+          ssh -o StrictHostKeyChecking=no -i private_key ${USER_NAME}@${HOSTNAME} '
+            cd amazon-cloudwatch-agent/integration/localstack &&
+            cd ls_tmp &&
+            openssl req -new -x509 -newkey rsa:2048 -sha256 -nodes -out snakeoil.pem -keyout snakeoil.key -config snakeoil.conf &&
+            cat snakeoil.key snakeoil.pem > server.test.pem &&
+            cat snakeoil.key > server.test.pem.key &&
+            cat snakeoil.pem > server.test.pem.crt &&
+            cat /etc/ssl/certs/ca-certificates.crt > original.pem &&
+            cat original.pem snakeoil.pem > combine.pem
+          '
+
+  StartLocalStack:
+    needs: [AddLocalStackKeys, Terraform]
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Start Local Stack on EC2
+        env:
+          PRIVATE_KEY: ${{ secrets.AWS_PRIVATE_KEY  }}
+          HOSTNAME: ${{ needs.Terraform.outputs.hostname }}
+          USER_NAME: ${{ secrets.USER_NAME  }}
+
+        run: |
+          echo "$PRIVATE_KEY" > private_key && chmod 600 private_key
+          ssh -o StrictHostKeyChecking=no -i private_key ${USER_NAME}@${HOSTNAME} '
+            cd amazon-cloudwatch-agent/integration/localstack &&
+            docker-compose up -d --force-recreate
+          '
+
+  RunIntegrationTest:
+    needs: [Terraform, StartLocalStack]
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run Integration Test
+        env:
+          PRIVATE_KEY: ${{ secrets.AWS_PRIVATE_KEY  }}
+          HOSTNAME: ${{ needs.Terraform.outputs.hostname }}
+          USER_NAME: ${{ secrets.USER_NAME  }}
+
+        run: |
+          echo "$PRIVATE_KEY" > private_key && chmod 600 private_key
+          ssh -o StrictHostKeyChecking=no -i private_key ${USER_NAME}@${HOSTNAME} '
+            cd amazon-cloudwatch-agent/integration/test &&
+            # Must run the test as sudo since it changes the /opt/aws/amazon-cloud-watch-agent files
+            sudo go test ./... -p 1 -v --tags=integration
+          '
+  TerraformDestroy:
+    if: always()
+    needs: [RunIntegrationTest, Terraform]
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: terraform
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Verify Terraform version
+        run: terraform --version
+
+      - name: Copy Terraform
+        env:
+          STATE: ${{ needs.Terraform.outputs.terraformState }}
+
+        run: |
+          echo "${STATE}" &&
+          echo "${STATE}" > terraform.tfstate
+
+      - name: Terraform init
+        env:
+          TERRAFORM_AWS_ACCESS_KEY_ID: ${{ secrets.TERRAFORM_AWS_ACCESS_KEY_ID }}
+          TERRAFORM_AWS_SECRET_ACCESS_KEY: ${{ secrets.TERRAFORM_AWS_SECRET_ACCESS_KEY }}
+        run: terraform init -var="aws_access_key=${TERRAFORM_AWS_ACCESS_KEY_ID}" -var="aws_secret_access_key=${TERRAFORM_AWS_SECRET_ACCESS_KEY}"
```

Comment:
Do you need to do an init before running destroy? Is the state file not enough?
> The terraform destroy command destroys all of the resources being managed by the current working directory and workspace, **using state data to determine which real world objects correspond to managed resources.**

https://www.terraform.io/docs/cli/run/index.html#destroying

Reply:
Jobs run on separate runners thus you still  need to run terraform init before destroy.